### ISSUE TITLE
Tried upgrading client-go and helm

### ISSUE THREAD

**Issue by JinoArch** [doitintl/kube-no-trouble#400]
Hello

I was just trying to update client-go to v0.20.4 and helm v3.5.4 and came across some issues, wonder if someone came across such 

- If I use helm 3.5.4 I see 
``` 
go get helm.sh/helm/v3@v3.5.4                                                                                                02:09:37 AM
go: downloading helm.sh/helm/v3 v3.5.4
go: helm.sh/helm/v3@v3.5.4 requires
	github.com/deislabs/oras@v0.10.0 requires
	github.com/docker/distribution@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```
This is a known issue but doesn't found a clear way to solve. [here](https://github.com/helm/helm/issues/9354)

- If I use helm 3.5.1 and try `go get `it works. But when I run the binary it shows:
```
# k8s.io/helm/pkg/storage/driver
../../../go/pkg/mod/k8s.io/helm@v2.17.0+incompatible/pkg/storage/driver/cfgmaps.go:66:36: not enough arguments in call to cfgmaps.impl.Get
	have (string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
	want (context.Context, string, "k8s.io/apimachinery/pkg/apis/meta/v1".GetOptions)
```
For this helm pulls apimachinery* version: 0.20.1
Looks like conflict b/w apimachinery and helm pack. But I used the the supported versions as per helm doc. Wonder how to solve this. Any input is appreciated :)

**Comment by kppullin** [comment]
You might try using the latest versions of these, if there's no specific reason you're looking to use v3.5.X.  The steps I list here works on my machine : ) - https://github.com/doitintl/kube-no-trouble/issues/391#issuecomment-1314611104

**Comment by stepanstipl** [comment]
Hi @JinoArch - as mentioned in #390 and also related #391 we currently have a couple of issues which will be solved by an upgrade of K8S `client-go`/`apimachinery`, which is currently blocked by Helm v2. This will also allow upgrading to a newer Helm v3.

Now I've tried this and as @kppullin mentioned, ended up with Helm `v3.10.2`, which seems to work and also seems to be the latest release [^1].

Is there a particular reason why you would be looking to use `v3.5.4`?  I'm asking as I'll be preparing PR with these upgrades (and dropping Helm v2). So if `v3.10.2` works for you, 🤞 we should have at least usable nightly build shortly. But I would still be interested in case there's a reason for `v3.5.4` I should be aware of. Cheers 👍 

[^1]: https://github.com/helm/helm/releases

**Comment by JinoArch** [comment]
hi :)  @stepanstipl & @kppullin .. The reason I was interested on v3.5.4 because I have a lab EKS cluster 1.21 and for accessing and playing around I uses custom client-go app of version 0.20.4. From helm doc I saw for 1.21 its good to use v3.5.4 as per [here](https://helm.sh/docs/topics/version_skew/)

But it looks like I have to update my lab EKS environment to latest to make use of helm 3.10.x

**Comment by JinoArch** [comment]
REQUEST: new tag and release is expected 🙂

**Comment by stepanstipl** [comment]
@JinoArch you can try the latest nightly build [^1], it should include the fix. It would be actually great if you could confirm the issue is resolved in this build. Other than that, I would like to fix one more thing (#402), and we will release it as `0.7.0`, 🤞 soon.

[^1]: https://github.com/doitintl/kube-no-trouble/releases/tag/nightly-0.6.0-15-g6424a0b

**Comment by JinoArch** [comment]
hello 

Sorry for the late reply. I was on vacation.

Yes it looks good. Latest tag for helm 3 ones (which you will release soon) and 0.6.0 or less with helm 2 and helm 3 

Thanks :)

**Comment by JinoArch** [comment]
@stepanstipl one more thing I noted while running this(nightly-build-0.6.0) and previous version(0.6.0)... is for helm3, it shows the ones which are already updated with latest api versions. I tried to find root cause in configmap and secrets but no luck. Can you help? I might be missing something!

```
>>> Deprecated APIs removed in 1.22 <<<
------------------------------------------------------------------------------------------
KIND      NAMESPACE     NAME       API_VERSION                 REPLACE_WITH (SINCE)
Ingress   jinos         jinos-ing  networking.k8s.io/v1beta1   networking.k8s.io/v1 (1.19.0)
```

Helm yaml present in cluster 

```
apiVersion: networking.k8s.io/v1
kind: Ingress
```

**Comment by stepanstipl** [comment]
So first, we now have a release with updated libs out - 0.7.0[^1]. 🎉 

As to the "already updated" - please open a new issue, as it's a different matter if it still persists. But I would suggest looking at the original helm manifests that were used to create these resources (one option is the `helm get manifest` command), as  retrieving the version directly by kubectl has no meaning, as the version is dictated by the client (I've previously explained this more in detail, see[^2][^3]).


[^1]: https://github.com/doitintl/kube-no-trouble/releases/tag/0.7.0
[^2]: https://github.com/doitintl/kube-no-trouble/issues/316#issuecomment-1245335960
[^3]: https://github.com/doitintl/kube-no-trouble/issues/309#issuecomment-1235376221
